test/quantization/test_quant_api.py

```diff
@@ -826,19 +826,23 @@ def test_config_deprecation(self):
             uintx_weight_only: (torch.uint4,),
         }
 
-        with warnings.catch_warnings(record=True) as _warnings:
-            # Call each deprecated API twice
-            for cls, args in deprecated_apis_to_args.items():
+        # Call each deprecated API twice
+        for cls, args in deprecated_apis_to_args.items():
+            with warnings.catch_warnings(record=True) as _warnings:
                 cls(*args)
                 cls(*args)
 
-            # Each call should trigger the warning only once
-            self.assertEqual(len(_warnings), len(deprecated_apis_to_args))
-            for w in _warnings:
-                self.assertIn(
-                    "is deprecated and will be removed in a future release",
-                    str(w.message),
-                )
+                # Each call should have at least one warning.
+                # Some of them can have two warnings - one for deprecation,
+                # one for moving to prototype
+                self.assertTrue(len(_warnings) > 0)
```

Comment:
I think we want to assert `len(_warnings) == 1` here, since we moved the warnings context manager outside the loop? E.g. if it's 2 then that means we've logged a warning each time we call the API, which could be very noisy